### PR TITLE
Backport 2.7: Platform setup terminate

### PR DESCRIPTION
**Pull request by k-stachowiak**
This is a backport of #1572

### CODE REVIEW

File: programs/ssl/ssl_pthread_server.c [hunk 1]
```diff
@@ -105,6 +105,10 @@ static void my_mutexed_debug( void *ctx, int level,
 }
 
 typedef struct {
+#if defined(MBEDTLS_PLATFORM_C)
+    mbedtls_platform_context platform_ctx;
+    mbedtls_threading_mutex_t platform_ctx_mutex;
```
Comment (mazimkhan) [inline programs/ssl/ssl_pthread_server.c]: Please see my comments https://github.com/ARMmbed/mbedtls/pull/1572#discussion_r182707551

File: programs/x509/cert_req.c [hunk 2]
```diff
@@ -138,11 +138,22 @@ int main( int argc, char *argv[] )
...
+#endif
     mbedtls_x509write_csr req;
     mbedtls_entropy_context entropy;
     mbedtls_ctr_drbg_context ctr_drbg;
     const char *pers = "csr example app";
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  !  mbedtls_platform_setup returned %d", ret );
```
Comment (RonEld) [inline programs/x509/cert_req.c]: shouldn't it be `-ret` for consistency?

File: programs/x509/cert_write.c [hunk 3]
```diff
@@ -230,6 +233,14 @@ int main( int argc, char *argv[] )
     mbedtls_ctr_drbg_context ctr_drbg;
     const char *pers = "crt example app";
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( mbedtls_platform_setup( &platform_ctx ) != 0 )
+    {
+        mbedtls_printf( "Failed initializing platform.\n" );
```
Comment (RonEld) [inline programs/x509/cert_write.c]: please change to ` failed\n  !  mbedtls_platform_setup returned %d` for consistency

File: programs/x509/crl_app.c [hunk 4]
```diff
@@ -72,6 +72,14 @@ int main( int argc, char *argv[] )
     mbedtls_x509_crl crl;
     int i;
     char *p, *q;
+#if defined(MBEDTLS_PLATFORM_C)
+    mbedtls_platform_context platform_ctx;
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  !  mbedtls_platform_setup returned %d\n\n", ret );
```
Comment (RonEld) [inline programs/x509/crl_app.c]: shouldn't it be `-ret` for consistency?

File: programs/x509/req_app.c [hunk 5]
```diff
@@ -72,6 +72,14 @@ int main( int argc, char *argv[] )
     mbedtls_x509_csr csr;
     int i;
     char *p, *q;
+#if defined(MBEDTLS_PLATFORM_C)
+    mbedtls_platform_context platform_ctx;
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  !  mbedtls_platform_setup returned %d\n\n", ret );
```
Comment (RonEld) [inline programs/x509/req_app.c]: shouldn't it be `-ret` for consistency?

File: programs/test/ssl_cert_test.c [hunk 6]
```diff
@@ -81,10 +81,21 @@ const char *client_private_keys[MAX_CLIENT_CERTS] =
...
+    mbedtls_platform_context platform_ctx;
+#endif
     mbedtls_x509_crt cacert;
     mbedtls_x509_crl crl;
     char buf[10240];
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  !  mbedtls_platform_setup returned %d\n\n", ret );
```
Comment (RonEld) [inline programs/test/ssl_cert_test.c]: shouldn't it be `-ret` for consistency?

File: programs/ssl/ssl_mail_client.c [hunk 7]
```diff
@@ -367,6 +370,14 @@ int main( int argc, char *argv[] )
     char *p, *q;
     const int *list;
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platform_setup returned %d\n", ret );
```
Comment (RonEld) [inline programs/ssl/ssl_mail_client.c]: shouldn't it be `-ret` for consistency?

File: programs/ssl/ssl_fork_server.c [hunk 8]
```diff
@@ -100,13 +100,23 @@ int main( void )
...
     mbedtls_ctr_drbg_context ctr_drbg;
     mbedtls_ssl_context ssl;
     mbedtls_ssl_config conf;
     mbedtls_x509_crt srvcert;
     mbedtls_pk_context pkey;
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed!  mbedtls_platform_setup returned %d\n\n", ret );
```
Comment (RonEld) [inline programs/ssl/ssl_fork_server.c]: shouldn't it be `-ret` for consistency?

File: programs/ssl/ssl_client1.c [hunk 9]
```diff
@@ -99,6 +102,13 @@ int main( void )
     /*
      * 0. Initialize the RNG and the session data
      */
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platform_setup returned %d\n", ret );
```
Comment (RonEld) [inline programs/ssl/ssl_client1.c]: shouldn't it be `-ret` for consistency?

File: programs/random/gen_random_ctr_drbg.c [hunk 10]
```diff
@@ -53,21 +53,39 @@ int main( int argc, char *argv[] )
...
+    mbedtls_platform_context platform_ctx;
+#endif
     mbedtls_ctr_drbg_context ctr_drbg;
     mbedtls_entropy_context entropy;
     unsigned char buf[1024];
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( "failed in mbedtls_platform_setup: %d\n", ret );
```
Comment (RonEld) [inline programs/random/gen_random_ctr_drbg.c]: shouldn't it be `-ret` for consistency?

File: programs/pkey/rsa_verify_pss.c [hunk 11]
```diff
@@ -61,11 +61,22 @@ int main( int argc, char *argv[] )
...
+#endif
     mbedtls_pk_context pk;
     unsigned char hash[32];
     unsigned char buf[MBEDTLS_MPI_MAX_SIZE];
     char filename[512];
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( "  ! mbedtls_platform_setup returned %d\n\n", ret );
```
Comment (RonEld) [inline programs/pkey/rsa_verify_pss.c]: shouldn't it be `-ret` for consistency?

File: programs/pkey/rsa_sign_pss.c [hunk 12]
```diff
@@ -70,6 +73,13 @@ int main( int argc, char *argv[] )
     const char *pers = "rsa_sign_pss";
     size_t olen = 0;
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platform_setup returned %d\n", ret );
```
Comment (RonEld) [inline programs/pkey/rsa_sign_pss.c]: shouldn't it be `-ret` for consistency?

File: programs/pkey/rsa_genkey.c [hunk 13]
```diff
@@ -70,6 +73,14 @@ int main( void )
     FILE *fpriv = NULL;
     const char *pers = "rsa_genkey";
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platform_setup returned %d\n", ret );
```
Comment (RonEld) [inline programs/pkey/rsa_genkey.c]: shouldn't it be `-ret` for consistency?

File: programs/pkey/mpi_demo.c [hunk 14]
```diff
@@ -50,6 +50,15 @@ int main( void )
     int ret;
     mbedtls_mpi E, P, Q, N, H, D, X, Y, Z;
 
+#if defined(MBEDTLS_PLATFORM_C)
+    mbedtls_platform_context platform_ctx;
+    if( mbedtls_platform_setup( &platform_ctx ) != 0 )
+    {
+        mbedtls_printf( "  Failed initializing platform.\n" );
```
Comment (RonEld) [inline programs/pkey/mpi_demo.c]: should we have a `ret` here?

File: programs/aes/crypt_and_hash.c [hunk 15]
```diff
@@ -99,6 +102,13 @@ int main( int argc, char *argv[] )
       off_t filesize, offset;
 #endif
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( mbedtls_platform_setup( &platform_ctx ) != 0 )
```
Comment (RonEld) [inline programs/aes/crypt_and_hash.c]: should we set `ret` here and modify log for consistency?

File: programs/hash/generic_sum.c [hunk 16]
```diff
@@ -172,7 +172,14 @@ int main( int argc, char *argv[] )
     int ret, i;
     const mbedtls_md_info_t *md_info;
     mbedtls_md_context_t md_ctx;
-
+#if defined(MBEDTLS_PLATFORM_C)
+    mbedtls_platform_context platform_ctx;
+    if( mbedtls_platform_setup( &platform_ctx ) != 0 )
+    {
+        mbedtls_fprintf( stderr, "Failed initializing platform\n" );
```
Comment (RonEld) [inline programs/hash/generic_sum.c]: should we set `ret` here and modify log for consistency?

File: programs/hash/hello.c [hunk 17]
```diff
@@ -52,21 +52,38 @@ int main( void )
     unsigned char digest[16];
     char str[] = "Hello, world!";
 
+#if defined(MBEDTLS_PLATFORM_C)
+    mbedtls_platform_context platform_ctx;
+    if( mbedtls_platform_setup( &platform_ctx ) != 0 )
+    {
+        mbedtls_printf( "  Failed initializing platform.\n" );
```
Comment (RonEld) [inline programs/hash/hello.c]: should we set `ret` here and modify log for consistency?

File: programs/pkey/dh_client.c [hunk 18]
```diff
@@ -80,12 +80,22 @@ int main( void )
...
     mbedtls_entropy_context entropy;
     mbedtls_ctr_drbg_context ctr_drbg;
     mbedtls_rsa_context rsa;
     mbedtls_dhm_context dhm;
     mbedtls_aes_context aes;
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platform_setup returned %d\n", ret );
```
Comment (RonEld) [inline programs/pkey/dh_client.c]: shouldn't it be `-ret` for consistency?

File: programs/pkey/dh_genprime.c [hunk 19]
```diff
@@ -78,15 +81,23 @@ int main( int argc, char **argv )
     int i;
     char *p, *q;
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( mbedtls_platform_setup( &platform_ctx ) != 0 )
+    {
+        mbedtls_printf( "Failed initializing platform.\n" );
```
Comment (RonEld) [inline programs/pkey/dh_genprime.c]: should we set `ret` here and modify log for consistency?

File: programs/pkey/dh_server.c [hunk 20]
```diff
@@ -88,6 +91,13 @@ int main( void )
 
     mbedtls_mpi N, P, Q, D, E;
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platform_setup returned %d\n", ret );
```
Comment (RonEld) [inline programs/pkey/dh_server.c]: shouldn't it be `-ret` for consistency?

File: programs/pkey/ecdh_curve25519.c [hunk 21]
```diff
@@ -60,6 +63,13 @@ int main( int argc, char *argv[] )
     ((void) argc);
     ((void) argv);
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( ret = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platform_setup returned %d\n", ret );
```
Comment (RonEld) [inline programs/pkey/ecdh_curve25519.c]: shouldn't it be `-ret` for consistency?

File: programs/pkey/ecdsa.c [hunk 22]
```diff
@@ -109,6 +112,13 @@ int main( int argc, char *argv[] )
     const char *pers = "ecdsa";
     ((void) argv);
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( mbedtls_platform_setup( &platform_ctx ) != 0 )
+    {
+        mbedtls_printf( "  Failed initializing platform.\n" );
```
Comment (RonEld) [inline programs/pkey/ecdsa.c]: should we set `ret` here and modify log for consistency?

**Comment by AndrzejKurek** [comment]
Added comments as requested. @RonEld - please re-review.

**Comment by sbutcher-arm** [comment]
Could you please rebase to remove the conflicts?

**Comment by sbutcher-arm** [comment]
@RonEld / @AndrzejKurek - Can you please review this?

**Comment by AndrzejKurek** [comment]
@sbutcher-arm I have also commited to this PR, and I'm not sure if I'm the best person to review it.

**Comment by sbutcher-arm** [comment]
@k-stachowiak - Can you please rebase? You have conflicts.

**Comment by k-stachowiak** [comment]
@sbutcher-arm I have rebased this PR.

**Comment by k-stachowiak** [comment]
retest

**Comment by gilles-peskine-arm** [comment]
The 2.7 branch has reached its end of life, so I am closing this pull request.

**Pull request closed without merging**